src/main/java/com/expediagroup/hiveberg/IcebergObjectInspectorGenerator.java

```diff
@@ -0,0 +1,83 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.hiveberg;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.typeinfo.ListTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.MapTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.StructTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Types;
+
+class IcebergObjectInspectorGenerator {
+
+  protected ObjectInspector createObjectInspector(Schema schema) throws Exception {
+    List<String> columnNames = setColumnNames(schema);
+    IcebergSchemaToTypeInfo converter = new IcebergSchemaToTypeInfo();
+    List<TypeInfo> columnTypes = converter.getColumnTypes(schema);
+
+    List<ObjectInspector> columnOIs = new ArrayList<>(columnTypes.size());
+    for(int i = 0; i < columnTypes.size(); i++) {
+      columnOIs.add(createObjectInspectorWorker(columnTypes.get(i)));
+    }
+    return ObjectInspectorFactory.getStandardStructObjectInspector(columnNames, columnOIs, null);
+  }
+
+  protected ObjectInspector createObjectInspectorWorker(TypeInfo typeInfo) throws Exception {
+    ObjectInspector.Category typeCategory = typeInfo.getCategory();
+
+    switch(typeCategory) {
+      case PRIMITIVE:
+        PrimitiveTypeInfo pti = (PrimitiveTypeInfo)typeInfo;
+        return PrimitiveObjectInspectorFactory.getPrimitiveJavaObjectInspector(pti);
+      case LIST:
+        ListTypeInfo ati = (ListTypeInfo)typeInfo;
+        return ObjectInspectorFactory
+            .getStandardListObjectInspector(createObjectInspectorWorker(ati.getListElementTypeInfo()));
+      case MAP:
+        MapTypeInfo mti = (MapTypeInfo)typeInfo;
+        return ObjectInspectorFactory.getStandardMapObjectInspector(
+            createObjectInspectorWorker(mti.getMapKeyTypeInfo()),
+            createObjectInspectorWorker(mti.getMapValueTypeInfo()));
+      case STRUCT:
+        StructTypeInfo sti = (StructTypeInfo)typeInfo;
+        List<ObjectInspector> ois = new ArrayList<>(sti.getAllStructFieldTypeInfos().size());
+        for(TypeInfo structTypeInfos : sti.getAllStructFieldTypeInfos()) {
+          ois.add(createObjectInspectorWorker(structTypeInfos));
+        }
+        return ObjectInspectorFactory.getStandardStructObjectInspector(sti.getAllStructFieldNames(), ois);
+      default:
+        throw new Exception("Couldn't create Object Inspector for category: '" + typeCategory + "'");
```

Comment:
Should we make this a more specific `UnsupportedOperationException` like you do below?